plugin/src/ornaments/lifting.ml

```diff
@@ -76,10 +76,14 @@ let promotion_term_to_types env sigma trm =
  * True if forwards, false if backwards
  *)
 let direction_cached env from_typ promote k sigma : bool state =
-  let promote = unwrap_definition env promote in
-  let promote_env = zoom_env zoom_lambda_term env promote in
-  let sigma, promote_typ = infer_type promote_env sigma (mkRel 1) in
-  sigma, is_or_applies from_typ promote_typ
+  match k with
+  | UnpackSigma ->
+     sigma, true (* TODO!!! *)
```

Comment:
Very important: detect the direction here, otherwise backward direction will never trigger